src/app/examples/modal-example/first-embedded-modal-example/first-embedded-modal-example.component.ts

```diff
@@ -0,0 +1,34 @@
+import { EventData } from 'tns-core-modules/data/observable';
+import { Component, ViewContainerRef } from '@angular/core';
+
+import { ModalService } from '~/kirby/components/modal/services/modal-service';
+import { EmbeddedModalComponent } from '~/kirby/components/modal/embedded-modal.component';
+import { ModalConfig } from '~/kirby/components/modal/config/modal-config';
+import { SecondEmbeddedModalExampleComponent } from '../second-embedded-modal-example/second-embedded-modal-example.component';
+
+@Component({
+  templateUrl: './first-embedded-modal-example.component.html',
+  providers: [ModalService],
+})
+export class FirstEmbeddedModalExampleComponent implements EmbeddedModalComponent {
+  uid: number;
+
+  constructor(private modalService: ModalService, private vcRef: ViewContainerRef) {}
+
+  async openSecondModal(_args: EventData) {
+    const config: ModalConfig = {
+      title: 'Second Embedded Modal',
+      titleHorizontalAlignment: 'left',
+      closeIcon: 'arrow',
+      component: SecondEmbeddedModalExampleComponent,
+    };
+
+    // supposing no callback needed for the second component
+    await this.modalService.showModal(config, this.vcRef);
```

Comment:
I'd prefer if the uid was explicitly returned to caller instead:
```suggestion
    this.secondModalUId = await this.modalService.showModal(config, this.vcRef);
```

Reply:
I don't see why you would need the modal ID in the caller component. Typically your caller component will appear behind the modal which is going to open on "showModal(...)", so you won't be able to use it much. Nevertheless, this functionality is now implemented.